### PR TITLE
View datastream feature

### PR DESCRIPTION
**Pull request by diboy2**

### CODE REVIEW

File: client/hooks/useDatastreamOperation.ts [hunk 1]
```diff
@@ -93,10 +93,35 @@ const useDatastreamOperation = () => {
         }
     };
 
+    const viewDatastream = async (): Promise<{ data: string; mimeType: string; }> => {
+        try {
+            const response =  await fetchBlob(viewObjectDatastreamUrl(currentPid, activeDatastream));
+            const mimeType: string = response?.headers?.get("Content-Type").split(";")[0];
+            const data = mimeType.match(/text\/[-+.\w]+/) ?
```
Comment (demiankatz) [inline client/hooks/useDatastreamOperation.ts]: I find ternaries more readable like

```
x
? y
: z
```

than

```
x ?
y :
z
```

I think that having the operators in front makes it a little more obvious what is going on at a glance -- though of course that's a matter of opinion.

**Review by demiankatz — COMMENTED** [review]
This looks good to me, though I had one very minor suggestion.

I also know you were planning to do a little more work on flac/tiff behavior, so I'll wait to approve this until that final piece is finished.

File: client/hooks/useDatastreamOperation.ts (hunk 1 above)
Comment (diboy2) [reply client/hooks/useDatastreamOperation.ts]: Made change.

**Comment by demiankatz** [comment]
Thanks, @diboy2 -- I know you have a few more changes pending, so please request another review when you're ready to finalize this.

**Review by demiankatz — APPROVED** [review]
Looks good to me!

**Pull request merged**